src/GridLayoutProvider.ts

```diff
@@ -0,0 +1,74 @@
+
+import {Dimension, Layout, LayoutProvider, LayoutManager} from "recyclerlistview";
+import GridLayoutManager from "./GridLayoutManager";
+export default class GridLayoutProvider extends LayoutProvider {
+  private _getDimensionForIndex: (index: number) => number;
+  private _getSpanForIndex: (index: number) => number;
+  private _setMaxSpan: () => number;
+  private _renderWindowSize: Dimension | undefined;
+  private _isHorizontal: boolean | undefined;
+  constructor(
+    getLayoutTypeForIndex: (index: number) => string | number,
+    getDimensionForIndex: (index: number) => number,
+    getSpanForIndex: (index: number) => number,
+    setMaxSpan: () => number,
```

Comment:
Why is this still a method?

Reply:
This is for the GridLayoutProvider. GridLayoutManager gets the value in the constructor directly.